dbt/adapters/spark_livy/livysession.py

```diff
@@ -37,6 +37,102 @@
 
 DEFAULT_POLL_WAIT = 2
 
+class LivySession:
+    def __init__(self, connect_url, auth, headers, verify_ssl_certificate):
+        self.connect_url = connect_url
+        self.auth = auth
+        self.headers = headers
+        self.session_id = None
+        self.verify_ssl_certificate = verify_ssl_certificate
+
+    def __enter__(self):
+        return self
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_val: Exception | None,
+        exc_tb: TracebackType | None,
+    ) -> bool:
+        self.delete_session()
+        return True
+
+    def create_session(self, data):
+        # Create sessions
+        response = None
+        try:
+            response = requests.post(
+                       self.connect_url + '/sessions', data=json.dumps(data),
+                       headers=self.headers,
+                       auth=self.auth,
+                       verify=self.verify_ssl_certificate
+            )
+            response.raise_for_status()
+        except requests.exceptions.ConnectionError as c_err:
+            print("Connection Error :", c_err)
+        except requests.exceptions.HTTPError as h_err:
+            print("Http Error: ", h_err)
+        except requests.exceptions.Timeout as t_err:
+            print("Timeout Error: ", t_err)
+        except requests.exceptions.RequestException as a_err:
+            print("Authorization Error: ", a_err)
+
+        if response is None:
+            raise Exception("Invalid response from livy server")
+
+        self.session_id = None
+        try:
+            self.session_id = str(response.json()['id'])
+        except requests.exceptions.JSONDecodeError as json_err:
+            raise Exception("Json decode error to get session_id") from json_err
+
+        # Wait for started state
+        while True:
+            res = requests.get(
+                self.connect_url + '/sessions/' + self.session_id + '/state',
+                headers=self.headers,
+                auth=self.auth,
+                verify=self.verify_ssl_certificate
+            ).json()
+
+            if res['state'] == 'idle':
+                break
+            if res['state'] == 'dead':
+                print("ERROR, cannot create a livy interactive session")
+                raise dbt.exceptions.FailedToConnectException(
+                        'failed to connect'
+                    )
+                return
+
+            time.sleep(DEFAULT_POLL_WAIT)
+
+        logger.debug(f"Creating new livy session: {self.session_id}")
+
+        return self.session_id
+
+    def delete_session(self):
```

Comment:
we need to handle delete_session as it is being called at multiple places.

Reply:
any exception is logged when trying to delete a session